### PR TITLE
Move the integration tests to a submodule

### PR DESCRIPTION
**Pull request by roehrich-hpe**

### CODE REVIEW

**Comment by roehrich-hpe** [comment]
The new nnf-integration-test repo is here: https://github.com/NearNodeFlash/nnf-integration-test
That repo preserves all of the history that was in the `nnf-deploy/test` directory structure, using these instructions: https://www.johno.com/move-directory-between-repos-with-git-history .  Note that `git filter-branch` prints out warnings about it being less-than-perfect, so there's some buyer-beware here as well.

File: go.mod [hunk 1]
```diff
@@ -2,39 +2,25 @@ module github.com/NearNodeFlash/nnf-deploy
 
 go 1.19
 
-replace (
-	github.com/HewlettPackard/dws => ./dws
-	github.com/NearNodeFlash/lustre-fs-operator => ./lustre-fs-operator
-	github.com/NearNodeFlash/nnf-dm => ./nnf-dm
-	github.com/NearNodeFlash/nnf-sos => ./nnf-sos
-)
+replace github.com/HewlettPackard/dws => ./dws
```
Comment (bdevcich-hpe) [inline go.mod:5]: should dws be removed here too?

**Comment by bdevcich-hpe** [comment]
Depending on if we are going to release nnf-integration-test, we should make sure the release docs include it.

**Review by bdevcich-hpe — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (roehrich-hpe) [reply go.mod:5]: main.go uses `dwsv1alpha2.SystemConfiguration{}`

**Review by matthew-richerson — APPROVED** [review]

**Pull request merged**